### PR TITLE
Sync batch operations api

### PR DESCRIPTION
**Pull request by rodrigozhou**
## What was changed

Batch operations were added in API in https://github.com/temporalio/api/pull/197.
This PR syncs generated files affected by API changes.

## Why?

Unblock other changes that depends on recent API changes.

## Checklist

1. How was this tested:

`make`

### CODE REVIEW

File: internal/cmd/generateinterceptor/main.go
Comment (rodrigozhou) [inline internal/cmd/generateinterceptor/main.go:79]: I added the new API module here for the generator.

File: go.mod
Comment (cretz) [inline go.mod:18]: Is this implemented in the server? If not, I think we should wait until then to merge this. Also, I wonder if once implemented in the server we should tag the API since this is such a big change.

Comment (rodrigozhou) [reply go.mod:18]: The server has dummy implementations returning `Unimplemented` error.

Comment (cretz) [reply go.mod:18]: Why not wait until implemented there to add here? Can reference a local SDK perfect path during development.

Comment (rodrigozhou) [reply go.mod:18]: I added those to unblock myself since I made changes to the API (same with this PR). Cc: @yux0

Comment (yiminc) [reply go.mod:18]: The server implementation PR (for batch API) will be land soon, but we should not block this PR just because server currently return unimplemented as this API is not used anywhere anyway. Server also want to use UpsertMemo feature which is currently blocked on this PR.

Comment (rodrigozhou) [reply go.mod:18]: Just to be clear, this PR only updates the generated files (`client/service_proxy.go` and `converter/grpc_interceptor.go`). It does not create/expose an API that call those unimplemented API endpoints.

Comment (cretz) [reply go.mod:18]: > but we should not block this PR just because server currently return unimplemented

Why not? This is a user-facing API update to a non-tagged API version.

> Server also want to use UpsertMemo feature which is currently blocked on this PR.

This should not block development. I can show how to use `go.mod` replace clauses to reference a local form of this while you develop. If server wants to use `UpsertMemo`, that's blocked by #883 where I've also requested we have real tests to even confirm it works. All of this work can be done locally.

Comment (rodrigozhou) [reply go.mod:18]: I meant that this PR blocks merging the `UpsertMemo` PR. I've done all the  local tests, and also almost done with writing the integration tests there.

Comment (cretz) [reply go.mod:18]: :+1: Yeah, integration tests means things are working which works for me. I'd happily approve/merge. Curious, why does this PR block that one?

Comment (rodrigozhou) [reply go.mod:18]: The changes to API related to batch operations affect the generated files `grpc_interceptor.go` and `service_proxy.go`. This PR just re-generates those files. It has nothing to do with `UpsertMemo`, but since my changes also affect those files, I thought it would better to have a PR doing an update for batch related stuff first instead of altogether.

Comment (cretz) [reply go.mod:18]: Sure, you're call, updating API is updating API, could do in same PR or other.

So I can merge this. My concern is that we are giving new public API to end users that is not tagged and is not implemented. But as we've done with other API changes, we can just hope they don't try to use it yet.

**Review by cretz — APPROVED** [review]
Approving though I am not a fan of untagged, unimplemented API in SDK `master`, but :shrug:. You'll need to update the PR w/ latest master and then can merge after CI runs.

**Pull request merged**
